tasks/Task.cpp

```diff
@@ -220,6 +250,72 @@ void Task::outputAnalog()
     m_driver->resetAnalogInputTracking();
     m_driver->resetConvertedAnalogInputTracking();
 }
+
+bool Task::handleDigitalCommand()
+{
+    linux_gpios::GPIOState digital_cmd;
+    if (_digital_cmd.read(digital_cmd) != RTT::NewData) {
```

Comment:
There's a bit of a race here, since we're not reading the state of the I/O all the time.

I would just check for NoData, even more so since you're not doing anything if the states match.